src/fsharp/NameResolution.fs

```diff
@@ -4208,17 +4212,27 @@ let ResolveCompletionsInTypeForItem (ncenv: NameResolver) nenv m ad statics ty (
             | Item.MethodGroup _ ->
                 // REVIEW: add a name filter here in the common cases?
                 let minfos = 
-                    let minfos =
-                        AllMethInfosOfTypeInScope ncenv.InfoReader nenv (None,ad) PreferOverrides m ty 
-                        |> List.filter minfoFilter
+                    let minfos = AllMethInfosOfTypeInScope ncenv.InfoReader nenv (None,ad) PreferOverrides m ty 
+                    if isNil minfos then [] else
+                     
+                    let suppressedMethNames = Zset.ofList String.order (pinfoMethNames @ einfoMethNames)
+                    let minfos = 
+                        minfos
+                        |> List.filter (minfoFilter suppressedMethNames)
         
+                    if isNil minfos then [] else
```

Comment:
@forki,

Whilst I applaud the desire to reduce the code change here,  I find the if (expr) then expr else idiom completely unreadable.

Can we indent the if then properly please.  Other than that, I think the changes look good.

Kevin

Reply:
We have that thing in multiple places.